File: src/data/courses/dsa/linked-lists/special-patterns/flatten-multilevel/code/java/FlattenMultilevel.java
// ============================================================
// Flatten Multilevel DLL
// ============================================================
class Solution{public Node flatten(Node head){
    Node curr=head;
    while(curr!=null){
        if(curr.child!=null){
            Node child=curr.child,tail=child;
            while(tail.next!=null)tail=tail.next;
            tail.next=curr.next;if(curr.next!=null)curr.next.prev=tail;
            curr.next=child;child.prev=curr;curr.child=null;
        }
        curr=curr.next;
    }
    return head;
}}

File: src/data/courses/dsa/linked-lists/special-patterns/flatten-multilevel/code/python/flatten_multilevel.py
# ============================================================
# Flatten Multilevel DLL
# ============================================================
class Solution:
    def flatten(self,head):
        curr=head
        while curr:
            if curr.child:
                child=curr.child;tail=child
                while tail.next:tail=tail.next
                tail.next=curr.next
                if curr.next:curr.next.prev=tail
                curr.next=child;child.prev=curr;curr.child=None
            curr=curr.next
        return head
